examples/word2vec/train_word2vec.py

```diff
@@ -0,0 +1,186 @@
+import time
+
+import numpy as np
+import six.moves.cPickle as pickle
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+import chainer.optimizers as O
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--gpu', '-g', default=-1, type=int,
+                    help='GPU ID (negative value indicates CPU)')
+parser.add_argument('--unit', '-u', default=100, type=int,
+                    help='number of units')
+parser.add_argument('--window', '-w', default=5, type=int,
+                    help='window size')
+parser.add_argument('--batchsize', '-b', type=int, default=100,
+                    help='learning minibatch size')
+parser.add_argument('--epoch', '-e', default=10, type=int,
+                    help='number of epochs to learn')
+parser.add_argument('--model', '-m', choices=['skipgram', 'cbow'],
+                    default='skipgram',
+                    help='model type ("skipgram", "cbow")')
+parser.add_argument('--out-type', '-o', choices=['hsm', 'ns', 'original'],
+                    default='hsm',
+                    help='output model type ("hsm": hierarchical softmax, '
+                    '"ns": negative sampling, "original": no approximation)')
+args = parser.parse_args()
+
+print('GPU: {}'.format(args.gpu))
+print('# unit: {}'.format(args.unit))
+print('Window: {}'.format(args.window))
+print('Minibatch-size: {}'.format(args.batchsize))
+print('# epoch: {}'.format(args.epoch))
+print('Training model: {}'.format(args.model))
+print('Output type: {}'.format(args.out_type))
+print('')
+
+
+def continuous_bow(dataset, position):
+    h = None
+
+    # use random window size in the same way as the original word2vec
+    # implementation.
+    w = np.random.randint(args.window - 1) + 1
+    for offset in range(-w, w + 1):
+        if offset == 0:
+            continue
+        d = dataset[position + offset]
+        if args.gpu >= 0:
+            d = cuda.to_gpu(d)
+        x = chainer.Variable(d)
+        e = model.embed(x)
+        h = h + e if h is not None else e
+
+    d = dataset[position]
+    if args.gpu >= 0:
+        d = cuda.to_gpu(d)
+    t = chainer.Variable(d)
+    return loss_func(h, t)
+
+
+def skip_gram(dataset, position):
+    d = dataset[position]
+    if args.gpu >= 0:
+        d = cuda.to_gpu(d)
+    t = chainer.Variable(d)
+
+    # use random window size in the same way as the original word2vec
+    # implementation.
+    w = np.random.randint(args.window - 1) + 1
+    loss = None
+    for offset in range(-w, w + 1):
+        if offset == 0:
+            continue
+        d = dataset[position + offset]
+        if args.gpu >= 0:
+            d = cuda.to_gpu(d)
+        x = chainer.Variable(d)
+        e = model.embed(x)
+
+        loss_i = loss_func(e, t)
+        loss = loss_i if loss is None else loss + loss_i
+
+    return loss
+
+
+if args.gpu >= 0:
+    cuda.init(args.gpu)
+
+index2word = {}
+word2index = {}
+counts = collections.defaultdict(lambda: 0)
```

Comment:
``int`` can be used instead of ``lambda: 0``. You can also use collections.Counter.